### PR TITLE
feat: make starter html layout optional

### PR DESCRIPTION
**Pull request by vltansky**
```html
    <section class="swiper-container">
      <div class="swiper-slide">Slide 1</div>
      <div class="swiper-slide">Slide 2</div>
      <div class="swiper-slide">Slide 3</div>
      <div class="swiper-slide">Slide 4</div>
      <div class="swiper-slide">Slide 5</div>
      <div class="swiper-slide">Slide 6</div>
      <div class="swiper-slide">Slide 7</div>
      <div class="swiper-slide">Slide 8</div>
      <div class="swiper-slide">Slide 9</div>
      <div class="swiper-slide">Slide 10</div>
    </section>
```

The downside is probably CLS. But we can recommend to use full structure for better performance?

Not sure if it should target `v7` branch or just `master`?

### CODE REVIEW

File: src/components/scrollbar/scrollbar.js [hunk 1]
```diff
@@ -280,9 +280,12 @@ const Scrollbar = {
   },
   init() {
     const swiper = this;
-    if (!swiper.params.scrollbar.el) return;
     const { scrollbar, $el: $swiperEl } = swiper;
+    swiper.params.scrollbar = createElementIfNotDefined($swiperEl, swiper.params.scrollbar, [
```
[the remark this answers is not in the document]
Comment (vltansky) [reply src/components/scrollbar/scrollbar.js]: fixed

File: src/components/pagination/pagination.js [hunk 2]
```diff
@@ -248,6 +253,9 @@ const Pagination = {
   },
   init() {
     const swiper = this;
+    swiper.params.pagination = createElementIfNotDefined(swiper.$el, swiper.params.pagination, [
```
[the remark this answers is not in the document]
Comment (vltansky) [reply src/components/pagination/pagination.js]: fixed

File: src/components/navigation/navigation.js [hunk 3]
```diff
@@ -51,6 +51,11 @@ const Navigation = {
   init() {
     const swiper = this;
     const params = swiper.params.navigation;
+
+    swiper.params.pagination = createElementIfNotDefined(swiper.$el, swiper.params.pagination, [
```
[the remark this answers is not in the document]
Comment (vltansky) [reply src/components/navigation/navigation.js]: fixed

File: src/utils/utils.js [hunk 4]
```diff
@@ -146,6 +148,19 @@ function classesToSelector(classes = '') {
     .replace(/ /g, '.')}`;
 }
 
+function createElementIfNotDefined($container, params, arr) {
+  if (params.createElements) {
+    arr.forEach((item) => {
+      const [key, className] = item;
+      if (!params[key]) {
+        $container.append(`<div class="${className}"></div>`);
+        params[key] = $container.find(`.${className}`);
```
[the remark this answers is not in the document]
Comment (vltansky) [reply src/utils/utils.js:157]: done

File: src/components/core/core-class.js [hunk 5]
```diff
@@ -421,14 +421,20 @@ class Swiper {
...
-    let $wrapperEl;
-    if (el && el.shadowRoot && el.shadowRoot.querySelector) {
-      $wrapperEl = $(el.shadowRoot.querySelector(`.${swiper.params.wrapperClass}`));
-      // Children needs to return slot items
-      $wrapperEl.children = (options) => $el.children(options);
-    } else {
-      $wrapperEl = $el.children(`.${swiper.params.wrapperClass}`);
+    let $wrapperEl = getWrapper();
+    if ($wrapperEl.length === 0 && swiper.params.createElements) {
+      $el[0].innerHTML = `<div class="${swiper.params.wrapperClass}">${$el[0].innerHTML}</div>`;
```
[the remark this answers is not in the document]
Comment (vltansky) [reply src/components/core/core-class.js:446]:
```js

    // Find Wrapper
    let $wrapperEl = getWrapper();
    if ($wrapperEl.length === 0 && swiper.params.createElements) {
      const document = getDocument();
      const wrapper = document.createElement('div');
      wrapper.className = swiper.params.wrapperClass;
      $(el).append(wrapper);

      swiper.slides.each((slideEl) => {
        $(wrapper).append(slideEl);
      });

      $el[0].innerHTML = `<div class="${swiper.params.wrapperClass}">${$el[0].innerHTML}</div>`;
      $wrapperEl = getWrapper();
    }
```
is it looks good?

**Review by nolimits4web — APPROVED** [review]

File: src/types/swiper-class.d.ts [hunk 6]
```diff
@@ -444,10 +444,10 @@ interface Swiper extends SwiperClass<SwiperEvents> {
   keyboard: KeyboardMethods;
   lazy: LazyMethods;
   mousewheel: MousewheelMethods;
-  navigation: NavigationMethods;
-  pagination: PaginationMethods;
+  navigation: NavigationMethods | boolean;
```
Comment (dawidmalec7) [inline src/types/swiper-class.d.ts:447]: If someone uses TypeScript and used navigation methods in that way: `swiper.navigation.update();`. TypeScript will raise a error: "Property 'update' does not exist on type boolean | NavigationMethods"

**Pull request merged**